osu.Game/Screens/OnlinePlay/Playlists/PlaylistsPlayer.cs

```diff
@@ -65,7 +66,10 @@ private void load()
             {
                 failed = true;
 
-                Logger.Error(e, "Failed to retrieve a score submission token.\n\nThis may happen if you are running an old or non-official release of osu! (ie. you are self-compiling).");
+                if (e is WebException || string.IsNullOrEmpty(e.Message))
```

Comment:
This check doesn't actually work. I partially messed up (the exception thrown is actually an `APIException` - the *inner* is a `WebException`), but the larger issue is that `APIException` does not let consumers distinguish displayable errors (like in the "went over limit" case) from non-displayable ones (like the "no token" case).

For clarity, the "no token" case returns a 404 error code with the body of

```json
{"error": null}
```

A way to fix this client-side would be to store the `DisplayableError` that `APIException.Fail()` deserialises into the exception and then check if *that* message is null. Alternatively we could make web return an actual error message here.

Reply:
Yeah I think it would be more correct for web to return the actual error message, if feasible.

Reply:
This has been changed on osu-web end; will wait for the next release and apply fixes on our end following that.